test/common/upstream/subset_lb_test.cc

```diff
@@ -81,10 +81,48 @@ class TestMetadataMatchCriteria : public Router::MetadataMatchCriteria {
     return nullptr;
   }
 
+  Router::MetadataMatchCriteriaConstPtr
+  filterMatchCriteria(const std::set<std::string>& names) const override {
+    auto new_criteria = std::make_unique<TestMetadataMatchCriteria>(*this);
+    for (auto it = new_criteria->matches_.begin(); it != new_criteria->matches_.end();) {
+      if (names.count(it->get()->name()) == 0) {
+        it = new_criteria->matches_.erase(it);
+      } else {
+        it++;
+      }
+    }
+    return new_criteria;
+  }
+
 private:
   std::vector<Router::MetadataMatchCriterionConstSharedPtr> matches_;
 };
 
+class TestSubsetSelector : public SubsetSelector {
```

Comment:
Do we need this class? Seems like we could just use the real SubsetSelectImpl.

Reply:
We can use real class but I would have to add second constructor just for tests (the current one accepts protobuf parameters). LMK if adding second constructor for tests is OK. Alternatively I can add some helper function in the test for SubsetSelectorImpl creation.

Reply:
Added a helper function and switched to use real SubsetSelectorImpl.